site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,293 @@
+
+```bash
+$ echo $REPO_HOST
+  10.104.3.7:5000
+```
+
+Lets start by creating the needed directories:
+
+```bash
+$ mkdir -p nginx-bitnami-repo nginx-bitnami-repo/.imgpkg nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+```
+
+We can copy our CR YAMLs from the previous step in to the proper packages subdirectory. 
+
+```bash
+$ cp 1.0.0.yaml nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+$ cp pkgMetadata.yaml nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+```
+
+Now, let’s use kbld to record which package bundles are used:
+
+```bash
+$ kbld -f nginx-bitnami-repo/packages/ --imgpkg-lock-output nginx-bitnami-repo/.imgpkg/images.yml
+```
+
+We will push the bundle onto repository using `imgpkg`.
+
+```bash
+$ imgpkg push -b ${REPO_HOST}/packages/nginx-bitnami-repo:1.0.0 -f nginx-bitnami-repo
+```
+
+We can verify by checking the Docker registry catalog:
+
+```bash
+$ curl ${REPO_HOST}/v2/_catalog
+```
+
+--------------------------
+
+## Consuming Carvel Helm Package:
+
+**1. Install Package Repository**: Before installing a package, we have to create a PackageRepository first. A PackageRepository is a collection of packages which are available to install. 
+
+Save the below PackageRepository CR to a file named `repo.yaml`
+```yaml
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageRepository
+metadata:
+  name: simple-package-repository
+spec:
+  fetch:
+    imgpkgBundle:
+      image: ${REPO_HOST}/packages/nginx-bitnami-repo:1.0.0
+```
+
+```bash
+$ kapp deploy -a repo -f repo.yaml -y
+```
+
+After deploying, wait for the packageRepository description to become `Reconcile succeeded`. 
+
+```bash 
+$ kubectl get packagerepository
+```
+
+Now, we can see the list of package metadata's available
+
+```bash
+$ kubectl get packagemetadatas
+```
+
+**2. List Packages**:
+We can see the list of packages and their version's available for install 
+
+```bash 
+kubectl get packages
+```
+
+As we can see, our published nginx helm package is available for us to install.
+
+**3. Create Service Account**: To install the above package, we need to create default-ns-sa service account that give PackageInstall CR privileges to create resources in the default namespace
+
+```bash
+$ kapp deploy -a default-ns-rbac -f https://raw.githubusercontent.com/vmware-tanzu/carvel-kapp-controller/develop/examples/rbac/default-ns.yml -y
+```
+
+**4. Install the Package**: To install a carvel Package, we need to create PackageInstall Kuberentes resource. A Package Install will install the nginx helm package and its underlying resources on a Kubernetes cluster. A `PackageInstall` references a `Package`. Thus, we can create the `PackageInstall` yaml from the `Package`.
+
+We are providing our custom values via secret. 
+
+**NOTE**: If you are using minikube, for nginx service to be in `ACTIVE` state, start `minikube tunnel` in other window as services of LoadBalancer types do not come up otherwise in minikube.  
```

Comment:
nit:  "...in another window..."  reads better rather than just "other"

Reply:
Adopted